### DISCUSSION TITLE
$state type change from enum to string  in afterStateUpdated method when editing a form

### DISCUSSION THREAD

**Discussion by ahmetkocabiyik** [filamentphp/filament#10688] [Help]
When creating there is no issue with afterStateUpdated method but when editing giving an error like "Argument #3 ($state) must be of type App\Enums\PaymentTypeEnum, string given" When editing, i var_dump $state it looks like a string but when creating it looks enum. Why ?

```php
Forms\Components\Select::make("type")
                    ->options(PaymentTypeEnum::class)
                    ->default(PaymentTypeEnum::BANK_TRANSFER)
                    ->label("Payment Type")
                    ->live()
                    ->afterStateUpdated(function (Set $set, Get $get, PaymentTypeEnum $state){
                        dd($state);
                    })
                    ->columnSpanFull(),
```

**Comment by leandrocfe** [comment]
`mixed $state` ?

**Reply by ahmetkocabiyik** [reply]
It doesn't show any errors when i used mixed , but state must be enum. I want to understand why it change as string
